server/src/main/java/org/elasticsearch/action/bulk/TransportBulkAction.java

```diff
@@ -672,11 +672,12 @@ BulkRequest getBulkRequest() {
         void markCurrentItemAsDropped() {
             IndexRequest indexRequest = getIndexWriteRequest(bulkRequest.requests().get(currentSlot));
             failedSlots.set(currentSlot);
+            final String id = indexRequest.id() == null ? "auto-generated" : indexRequest.id();
```

Comment:
Maybe turn `auto-generated` into a constant?

Reply:
ok, added in 49e9b9d